Cython/Utility/TypeConversion.c

```diff
@@ -126,7 +126,8 @@ static CYTHON_INLINE Py_hash_t __Pyx_PyIndex_AsHash_t(PyObject*);
 #else
 #define __Pyx_PyNumber_Int(x) (PyInt_CheckExact(x) ? __Pyx_NewRef(x) : PyNumber_Int(x))
 #endif
-#define __Pyx_PyNumber_Float(x) (PyFloat_CheckExact(x) ? __Pyx_NewRef(x) : PyNumber_Float(x))
```

Comment:
Let's keep the PyFloat_CheckExact case here, deferring to a function call (inlined or not) only for the slower case. There may be a slight slowdown for non-string conversions (as all the other cases are called first), but we're creating a new object in that case anyway.

Reply:
I've refactored it to a macro with __Pyx_NewRef or a function call, as before. It still needs to move to it's own section I think because the function call depends on string formatting code, that we don't want to bring in unless needed.